src/nvim/getchar.c

```diff
@@ -4228,3 +4229,17 @@ static bool typebuf_match_len(const uint8_t *str, int *mlen)
   *mlen = i;
   return str[i] == NUL;  // matched the whole string
 }
+
+/// Retrieve the mapblock at the index either globally or for a certain buffer
+///
+/// @param  index  The index in the maphash[]
+/// @param  buf  The buffer to get the maphash from. NULL for global
+mapblock_T *get_maphash(int index, buf_T *buf)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
```

Comment:
You state that buffer is not NULL here, compiler was nice enough to determine this because I missed.